app/components/post_textbox/post_textbox.js

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import {injectIntl, intlShape} from 'react-intl';
+import {TextInput, TouchableHighlight, View} from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import {changeOpacity} from 'app/utils/colors';
+
+// import PaperClipIcon from './components/paper_clip_icon.js';
+
+const MAX_CONTENT_HEIGHT = 100;
+
+class PostTextbox extends React.Component {
+    static propTypes = {
+        currentUserId: React.PropTypes.string.isRequired,
+        teamId: React.PropTypes.string.isRequired,
+        channelId: React.PropTypes.string.isRequired,
+        rootId: React.PropTypes.string,
+        value: React.PropTypes.string.isRequired,
+        onChangeText: React.PropTypes.func.isRequired,
+        theme: React.PropTypes.object.isRequired,
+        actions: React.PropTypes.shape({
+            createPost: React.PropTypes.func.isRequired
+        }).isRequired,
+        intl: intlShape.isRequired
+    };
+
+    static defaultProps = {
+        rootId: ''
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
+
+        this.state = {
+            contentHeight: 0
+        };
+    }
+
+    handleContentSizeChange = (e) => {
+        this.setState({
+            contentHeight: e.nativeEvent.contentSize.height
+        });
+    }
+
+    sendMessage = () => {
+        if (this.props.value.trim().length === 0) {
+            return;
+        }
+
+        const post = {
+            user_id: this.props.currentUserId,
+            channel_id: this.props.channelId,
+            root_id: this.props.rootId,
+            parent_id: this.props.rootId,
+            message: this.props.value
+        };
+
+        this.props.actions.createPost(this.props.teamId, post);
+        this.props.onChangeText('');
+    }
+
+    render() {
+        const theme = this.props.theme;
+
+        let placeholder;
+        if (this.props.rootId) {
+            placeholder = this.props.intl.formatMessage({id: 'create_comment.addComment', defaultMessage: 'Add a comment...'});
+        } else {
+            placeholder = this.props.intl.formatMessage({id: 'create_post.write', defaultMessage: 'Write a message...'});
+        }
+
+        return (
+            <View>
+                <View
+                    style={{
+                        alignItems: 'flex-end',
+                        backgroundColor: theme.centerChannelBg,
+                        flex: 1,
+                        flexDirection: 'row',
+                        padding: 7
+                    }}
+                >
+                    {/*<TouchableHighlight
```

Comment:
I just left that there for when we add the ability to post files. The code and styling is there for the attachment button, so I just commented it out instead of removing it completely